### ISSUE TITLE
botbuilder-testing is missing install requirements

### ISSUE THREAD

**Issue by cas--** [microsoft/botbuilder-python#1627]
## Version

botbuilder-testing 4.12.0

## Describe the bug
While installing botbuilder-testing for CI I got errors about missing dependencies. 

## To Reproduce

1. `python3 -m venv .venv`
2. `. .venv/bin/activate`
3. `pip install -U pip wheel`
4. `pip install botbuilder-testing`
5. `python -c "from botbuilder.testing import DialogTestClient"`

First error is missing `pytest`:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/calum/sureswift/jell/jell-bot-teams-v2/.venv-test/lib/python3.8/site-packages/botbuilder/testing/__init__.py", line 6, in <module>
    from .storage_base_tests import StorageBaseTests
  File "/home/calum/sureswift/jell/jell-bot-teams-v2/.venv-test/lib/python3.8/site-packages/botbuilder/testing/storage_base_tests.py", line 26, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

6. `pip install pytest`
7. `python -c 'from botbuilder.testing import DialogTestClient'`

Next error is missing `botbuilder-azure`:
```python
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/calum/sureswift/jell/jell-bot-teams-v2/.venv-test/lib/python3.8/site-packages/botbuilder/testing/__init__.py", line 6, in <module>
    from .storage_base_tests import StorageBaseTests
  File "/home/calum/sureswift/jell/jell-bot-teams-v2/.venv-test/lib/python3.8/site-packages/botbuilder/testing/storage_base_tests.py", line 27, in <module>
    from botbuilder.azure import CosmosDbStorage
ModuleNotFoundError: No module named 'botbuilder.azure'
```

8. `pip install botbuilder-azure`
9. `python -c 'from botbuilder.testing import DialogTestClient'`

Command works!

## Expected behavior
No errors after installing botbuilder-testing and importing module

I do wonder if the requirement for pytest is not necessary, leaving the lib test-suite agnostic and could be refactored out?

**Comment by peterinnesmsft** [comment]
@axelsrz can you take a pass at this?

**Comment by lauren-mills** [comment]
Hi @cas--, this issue has been resolved in the PR #1637
